rfcs/features/0041-renderer-plugin/README.md

```diff
@@ -0,0 +1,209 @@
+- Explaining how vc-kit contributors and/or consumers should *think* about the
+feature, and how it should impact the way they use the ecosystem.
+- If applicable, provide sample error messages, deprecation warnings, or
+migration guidance.
+
+Some enhancement proposals may be more aimed at contributors (e.g. for
+consensus internals); others may be more aimed at consumers.
+
+## Reference
+
+Provide guidance for implementers, procedures to inform testing,
+interface definitions, formal function prototypes, error codes,
+diagrams, and other technical details that might be looked up.
+Strive to guarantee that:
+
+- Interactions with other features are clear.
+- Implementation trajectory is well defined.
+- Corner cases are dissected by example.
+
+## Drawbacks
+
+Why should we *not* do this?
+
+## Rationale and alternatives
+
+- Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not
+choosing them?
+- What is the impact of not doing this?
+
+## Prior art
+
+Discuss prior art, both the good and the bad, in relation to this proposal.
+A few examples of what this can include are:
+
+- Does this feature exist in other SSI ecosystems and what experience have
+their community had?
+- For other teams: What lessons can we learn from other attempts?
+- Papers: Are there any published papers or great posts that discuss this?
+If you have some relevant papers to refer to, this can serve as a more detailed
+theoretical background.
+
+This section is intended to encourage you as an author to think about the
+lessons from other implementers, provide readers of your proposal with a
+fuller picture. If there is no prior art, that is fine - your ideas are
+interesting to us whether they are brand new or if they are an adaptation
+from other communities.
+
+Note that while precedent set by other communities is some motivation, it
+does not on its own motivate an enhancement proposal here.
+
+## Unresolved questions
+
+- What parts of the design do you expect to resolve through the
+enhancement proposal process before this gets merged?
+- What parts of the design do you expect to resolve through the
+implementation of this feature before stabilization?
+- What related issues do you consider out of scope for this 
+proposal that could be addressed in the future independently of the
+solution that comes out of this doc?
+   
+## Implementations
+
+The following lists the implementations (if any) of this RFC. Please do a pull request to add your implementation. If the implementation is open source, include a link to the repo or to the implementation within the repo. Please be consistent in the "Name" field so that a mechanical processing of the RFCs can generate a list of all RFCs supported by a vc-kit implementation.
+
+*Implementation Notes* [may need to include a link to test results](/README.md#accepted).
+
+Name / Link | Implementation Notes
+--- | ---
+ | 
+
```

Comment:
- To implement the `renderer-plugin` , have you decided any specific plugin to use ? like `oa-renderer` or something else ?
 - Could you provide some more information about `WebRenderingTemplate2022 ` and `SvgRenderingHint2022 ` ? Since it relates on the format we gonna use to pass to the `renderer-plugin` ?

Reply:
those implementation notes are not for references to use to implement the plugin, when this is implemented, it will go there... 

they are two speculative approaches that do not have formal specifications at the moment. for now it is enough to know that one populates an svg template and returns a file object with relevant content type, and the other populates a html template and return file object with relevant content type

Reply:
@arpentnoir  , i have another questions 
 - Will this plugin be hosted or be like imported the library ?
 - What is the difference between the renderer-plugin and the oa-renderer ?

Reply:
it is not a deployable component, it is a plugin. an agent that has the plugin configured might be deployed somewhere, but that is out of scope of this task. 

the oa-renderer is an open attestation style decentralised renderer which needs to be hosted somewhere and is referenced by an open attestation credential. it is not a standard approach and is unique to oa credentials. 

the renderer-plugin is more generic, it is just an agent plugin, and as such it can be called wherever the agent is being used.